src/main/java/org/elasticsearch/index/shard/TranslogRecoveryPerformer.java

```diff
@@ -62,20 +57,30 @@ protected TranslogRecoveryPerformer(MapperService mapperService, MapperAnalyzer
         return mapperService.documentMapperWithAutoCreate(type); // protected for testing
     }
 
-    /*
+    /**
      * Applies all operations in the iterable to the current engine and returns the number of operations applied.
-     * This operation will stop applying operations once an opertion failed to apply.
+     * This operation will stop applying operations once an operation failed to apply.
+     *
+     * @param allowMappingUpdates true if mapping update should be accepted (but collected). Setting it to false will
+     *                            cause a {@link MapperException} to be thrown if an update
+     *                            is encountered.
      */
-    int performBatchRecovery(Engine engine, Iterable<Translog.Operation> operations) {
+    int performBatchRecovery(Engine engine, Iterable<Translog.Operation> operations, boolean allowMappingUpdates) {
```

Comment:
we only call this from one place so we don't need `boolean allowMappingUpdates` but can pass `false` directly?